internal/metrics/data_transfer.go

```diff
@@ -0,0 +1,88 @@
+package metrics
+
+import (
+	"fmt"
+	"reflect"
+	"sync/atomic"
+	"time"
+
+	"git.sr.ht/~spc/go-log"
+
+	"github.com/project-flotta/flotta-device-worker/internal/datatransfer"
+	"github.com/project-flotta/flotta-operator/models"
+)
+
+const (
+	dataTransferTargetName                     = "data transfer"
+	DefaultDataTransferMetricsScrapingInterval = 60
+)
+
+var defaultDataTransferMetricsConfiguration = models.ComponentMetricsConfiguration{Interval: DefaultDataTransferMetricsScrapingInterval}
+
+type DataTransferMetrics struct {
+	latestConfig atomic.Value
+	daemon       MetricsDaemon
+}
+
+func NewDataTransferMetrics(daemon MetricsDaemon) *DataTransferMetrics {
```

Comment:
Where this is called? I can't find any references in the files changed on this PR